decidim-elections/app/models/decidim/elections/election.rb

```diff
@@ -78,13 +78,18 @@ def minimum_answers?
         questions.all? { |question| question.answers.size > 1 }
       end
 
-      # Public: Checks if the election results are published and election finished
-      #
-      # Returns a boolean.
+      # # Public: Checks if the election results are published and election finished
+      # #
+      # # Returns a boolean.
       def results_published?
-        results = questions.collect(&:total_votes)
-        results = results.inject(0, :+)
-        results.positive? && finished?
+        bb_status == "results_published"
```

Comment:
We are using an enum for this, so you can ask this in an easier way: `results_published_bb_status?`.

The last part is because we are using a `_suffix: true` for the enum, so it creates this method automatically with the `bb_status` suffix. I don't know if this suffix is needed to avoid hiding another class method, but it would be nice to remove it. In that case, we shouldn't have to define any of this methods, as `results_published?` and `results?` will exist (and `key_ceremony?`, `ready?`, `vote?` and `tally?` too).

If this has any problem, I would say that we also could use something like `_prefix: :bb` instead of `_suffix: true`, to have `bb_results_published?` that sounds a bit better (at least for me 😅).

What do you think? Maybe @agustibr, has something to say here, as he created the enum. 😄

Reply:
I added the `suffix` option because of the `tally` (auto method) that raised an error.

I also find clearer the `_prefix: :bb`, so +1 to change it 😁